astropy/coordinates/spectral_coordinate.py

```diff
@@ -515,8 +460,14 @@ def radial_velocity(self):
         coordinate frame in that this calculates the radial velocity with
         respect to the *observer*, not the origin of the frame.
         """
-        return self._calculate_radial_velocity(self._observer, self._target,
-                                               as_scalar=True)
+        if self._observer is None or self._target is None:
+            if self._radial_velocity is None:
```

Comment:
Is it good to have this as a default? Or would `AttributeError` be more reasonable? Though I can see that if one wants there to be a (reasonable) default of 0, it is good to keep track of the fact that it is entered as a default, not an explicit value.

Reply:
Yes the idea was to differentiate between explicitly specified zero versus default though maybe that distinction is unimportant. I don't feel strongly about this, but I think raising an ``AttributeError`` would be too much.